File: regexlib/2021-5-15/python_re2_test_file/regexlib_3435.py
# 3435
# (ISBN[-]*(1[03])*[ ]*(: ){0,1})*(([0-9Xx][- ]*){13}|([0-9Xx][- ]*){10})
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"ISBN-10 : "*5000+"!1 _SLQ_1"

import re2 as re
from time import perf_counter

regex = """(ISBN[-]*(1[03])*[ ]*(: ){0,1})*(([0-9Xx][- ]*){13}|([0-9Xx][- ]*){10})"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "ISBN-10 : " * i * 10000 + "!1 _SLQ_1"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_3435.js
/* 3435
 * (ISBN[-]*(1[03])*[ ]*(: ){0,1})*(([0-9Xx][- ]*){13}|([0-9Xx][- ]*){10})
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"ISBN-10 : "*5000+"!1 _SLQ_1"
 */
var REGEX = /(ISBN[-]*(1[03])*[ ]*(: ){0,1})*(([0-9Xx][- ]*){13}|([0-9Xx][- ]*){10})/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + 'ISBN-10 : '.repeat(i*10000) + '!1 _SLQ_1'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}